src/Features/Core/Portable/SolutionCrawler/IDocumentTrackingServiceExtensions.cs

```diff
@@ -30,5 +33,43 @@ public static ImmutableArray<Document> GetVisibleDocuments(this IDocumentTrackin
                       .WhereNotNull()
                       .Distinct()
                       .ToImmutableArray();
+
+        public static Task DelayIfNonVisibleAsync(this IDocumentTrackingService service, Document document, TimeSpan timeSpan, CancellationToken cancellationToken)
+            => DelayIfNonVisibleAsync(service, ImmutableArray.Create(document), timeSpan, cancellationToken);
+
+        public static async Task DelayIfNonVisibleAsync(this IDocumentTrackingService service, ImmutableArray<Document> documents, TimeSpan timeSpan, CancellationToken cancellationToken)
```

Comment:
Either doc that this listens to visibility changes, and cancels, and/or rename to `DelayWhileNonVisibleAsync()`?